Compile error on Termux (Android)

- OS: Android (Termux)
 - vim-clap version: Latest from main.

**Describe the bug**

I get a compilation error when trying to upgrade:

```
error[E0282]: type annotations needed for `Box<_>`
  --> /data/data/com.termux/files/home/.cargo/registry/src/index.crates.io-6f17d22bba15001f/time-0.3.34/src/format_description/parse/mod.rs:83:9
   |
83 |     let items = format_items
   |         ^^^^^
...
86 |     Ok(items.into())
   |              ---- type must be known at this point
   |
help: consider giving `items` an explicit type, where the placeholders `_` are specified
   |
83 |     let items: Box<_> = format_items
   |              ++++++++

   Compiling utils v0.1.54 (/data/data/com.termux/files/home/.vim/bundle/vim-clap/crates/utils)
For more information about this error, try `rustc --explain E0282`.
error: could not compile `time` (lib) due to 1 previous error
warning: build failed, waiting for other jobs to finish...
```

**To Reproduce**
Steps to reproduce the behavior:

Just ran `Plugupdate`, also tried `cargo build --release --target aarch64-linux-android`

--- a/crates/filter/src/sequential_worker.rs
+++ b/crates/filter/src/sequential_worker.rs
@@ -448,7 +448,7 @@ mod tests {
 
                     item
                 })
-                .take(usize::max_value() >> 8),
+                .take(usize::MAX >> 8),
             ),
         )
         .unwrap()

--- a/crates/maple_config/doc_gen/src/main.rs
+++ b/crates/maple_config/doc_gen/src/main.rs
@@ -99,12 +99,15 @@ impl FieldInfo {
 }
 
 fn extract_doc_comment(attr: &Attribute) -> Option<String> {
-    if let Ok(MetaNameValue { path, value, .. }) = attr.meta.require_name_value() {
-        if let Expr::Lit(expr_lit) = value {
-            if let Lit::Str(comment) = &expr_lit.lit {
-                if path.is_ident("doc") {
-                    return Some(comment.value());
-                }
+    if let Ok(MetaNameValue {
+        path,
+        value: Expr::Lit(expr_lit),
+        ..
+    }) = attr.meta.require_name_value()
+    {
+        if let Lit::Str(comment) = &expr_lit.lit {
+            if path.is_ident("doc") {
+                return Some(comment.value());
             }
         }
     }

--- a/crates/maple_config/src/lib.rs
+++ b/crates/maple_config/src/lib.rs
@@ -458,7 +458,7 @@ pub struct SyntaxPluginConfig {
     /// Possible values:
     /// - `visual-lines`: Always render the visual lines only.
     /// - `entire-buffer-up-to-limit`: Render the entire buffer until
-    /// the buffer size exceeds the size limit (in bytes).
+    ///   the buffer size exceeds the size limit (in bytes).
     ///
     /// # Example
     ///

--- a/crates/maple_core/src/stdio_server/plugin/mod.rs
+++ b/crates/maple_core/src/stdio_server/plugin/mod.rs
@@ -126,10 +126,12 @@ pub trait ClapPlugin: ClapAction + Debug + Send + Sync + 'static {
 mod tests {
     #[derive(maple_derive::ClapPlugin)]
     #[clap_plugin(id = "plugin", actions = ["action1", "action2"])]
+    #[allow(dead_code)]
     struct TestPlugin;
 
     #[derive(maple_derive::ClapPlugin)]
     #[clap_plugin(id = "empty")]
+    #[allow(dead_code)]
     struct EmptyPlugin;
 
     #[test]

--- a/crates/maple_derive/tests/tests.rs
+++ b/crates/maple_derive/tests/tests.rs
@@ -1,3 +1,4 @@
+#![allow(dead_code)]
 use types::AutocmdEventType;
 
 #[async_trait::async_trait]

--- a/crates/printer/src/trimmer/v0.rs
+++ b/crates/printer/src/trimmer/v0.rs
@@ -4,7 +4,7 @@
 //! 1. `let g:clap_force_python = 1`.
 //! 2. open https://github.com/subspace/subspace/blob/c50bec907ab8ade923a2a0b4888f43bfc47e8a7f/polkadot/node/collation-generation/src/lib.rs
 //! 3. Type `sr` and then you'll see Neovim hang forever, have no idea&time to fix it
-//! properly therefore the old implementation are just kept.
+//!    properly therefore the old implementation are just kept.
 
 use super::AsciiDots;
 

--- a/crates/printer/src/truncation.rs
+++ b/crates/printer/src/truncation.rs
@@ -261,7 +261,7 @@ mod tests {
     fn test_grep_print() {
         // GrepResult { matched_item: MatchedItem { item: "crates/maple_core/src/paths.rs:198:31:let expected = \"~/.rustup/.../src/rust/library/alloc/src/string.rs\";", rank: [874, -30, -68, 0], indices: [68, 69, 77, 91, 92], display_text: None, output_text: None }, path: "/home/xlc/.vim/plugged/vim-clap/crates/maple_core/src/paths.rs", line_number: 198, line_number_start: 32, line_number_end: 35, column: 31, column_start: 36, column_end: 38 }, winwidth: 62, icon: Enabled(Grep)
         let line = r#"crates/maple_core/src/paths.rs:198:31:let expected = "~/.rustup/.../src/rust/library/alloc/src/string.rs";"#;
-        let mut items = vec![GrepResult {
+        let mut items = [GrepResult {
             matched_item: MatchedItem::new(
                 Arc::new(line.to_string()) as Arc<dyn ClapItem>,
                 [874, -30, -68, 0],

--- a/rust-toolchain.toml
+++ b/rust-toolchain.toml
@@ -1,2 +1,2 @@
 [toolchain]
-channel = "1.77"
+channel = "1.80"

